src/startofyear.c

```diff
@@ -36,27 +36,21 @@ SEXP do_startofyear (SEXP _from, SEXP _to, SEXP _origin)
   int *fromto = INTEGER(_fromto);
 
   int nyear[1] = { (to - from + 1) };
-  int leap[nyear[0]];
-
 
   // generate sequence of dates to work with
   fromto[0] = from;
   for(i=1; i < nyear[0]; i++) {
```

Comment:
I'm pretty sure we can combine this loop as well, but my naive attempt failed, the interaction between the two loops is somewhat subtle and I haven't turned on the full brain for this 😄

Reply:
I wouldn't worry about this. The benefit isn't worth the extra effort.